tests/pulses/time_extension_pulse_tests.py

```diff
@@ -59,4 +61,23 @@ def test_quantities(self):
         duration_extended = self.extend_both.duration.evaluate_in_scope(self.parameters)
         duration_summed = self.main_pt.duration.evaluate_in_scope(self.parameters)\
             + self.parameters['t_prior'] + self.parameters['t_posterior']
-        self.assertEqual(duration_extended, duration_summed)
\ No newline at end of file
+        self.assertEqual(duration_extended, duration_summed)
+        
+    def test_pad_to_usecase(self):
```

Comment:
What I mean by the previous comment is basically this, which gives a - in my opinion - quite convenient way of quickly extend waveforms with shorter and longer hold times guaranteed to match the hardware requirements

Reply:
I think a utility function like this would be sufficient

```python
def extend(self, *, total_duration = None, prepend = None, append = None) -> SequencePT:
    none_count = total_duration is None + prepend is None + append is None
    if none_count != 1:
        raise ValueError("Exactly two arguments must be specified.")
    # Return a SequencePT
```

Reply:
for that case of not concatenating the waveforms, yes; what about the single wf case? (Which i initially added this for)

Reply:
Ah, but why the SequencePT? Why represent the start and end as PulseTemplates? I would either have a PT that represents the time extention *without* introducing ConstantPTs in the front and in the back OR use SequencePT and add an atomic flag to it which encodes potential atomicity in the PT itself.

Reply:
seemed like the least programmatic effort to me, but that can of course be changed.